File: AndroidDefault/Android/UsingJava/View/RecyclerView/Java/RecycleView/app/src/main/java/com/example/recycleview/RecycleAdapter.java
package com.example.recycleview;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

public class RecycleAdapter extends RecyclerView.Adapter<RecycleAdapter.ViewHolder> {

    private String[] dataList;

    RecycleAdapter(String[] data){
        this.dataList = data;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
        View view = inflater.inflate(R.layout.recycle_view, viewGroup, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        String title = dataList[i];
        viewHolder.txtTitle.setText(title);
    }

    @Override
    public int getItemCount() {
        return dataList.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        ImageView imgIcon;
        TextView txtTitle;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            imgIcon = itemView.findViewById(R.id.recycle_view_icon);
            txtTitle = itemView.findViewById(R.id.recycle_view_txt);
        }
    }
}

File: AndroidDefault/Android/UsingKotlin/View/RecyclerView/RecyclerView/app/src/main/java/com/example/recyclerview/RecycleAdapter.kt
package com.example.recyclerview

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

class RecycleAdapter internal constructor(private val dataList: Array<String?>) :
    RecyclerView.Adapter<RecycleAdapter.ViewHolder>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val inflater = LayoutInflater.from(viewGroup.context)
        val view = inflater.inflate(R.layout.recycle_view, viewGroup, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val title = dataList[i]
        viewHolder.txtTitle.text = title
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        internal var imgIcon: ImageView
        internal var txtTitle: TextView

        init {
            imgIcon = itemView.findViewById(R.id.recycle_view_icon)
            txtTitle = itemView.findViewById(R.id.recycle_view_txt)
        }
    }
}
